### DISCUSSION TITLE
Wandb logging without storing images

### DISCUSSION THREAD

**Discussion by ameicler** [ultralytics/yolov5#7484] [Q&A]
Hi,

Thanks for this extensive and powerful yolov5 implementation.

I was wondering if there was an easy way (through parameters or commenting some part of code) to activate wandb logging but without storing (or sending) any image?

Thanks for your help.

**Comment by glenn-jocher** [comment]
@ameicler could you expand on your use case a little? Maybe @AyushExel can help you out.

**Comment by ameicler** [comment]
Sure. I would like to train object detection models with medical images and therefore not have any images stored/saved/sent to wandb platform (i.e. need to disable the boundingboxdebugger feature and maybe other).

Is there an easy way to do so?

**Comment by AyushExel** [comment]
@ameicler hey. You can set the `--bbox_interval` to be greater than the number of epochs to avoid logging any images in the media panel. But the other images like mosiacs and metrics images that are saved locally by the trainer will still be logged. Does this solve your problem? Happy to make tweaks if needed. Thanks!

@glenn-jocher on that note, do you think having a flag similar to `--nosave` to turn off media logging would be useful for users? Something like `--nomedia` ?

**Reply by glenn-jocher** [reply]
@AyushExel well it's an interesting idea for enterprise clients to address data privacy concerns. If we did a flag for this we'd probably want to eliminate the actual saving of the images in the first place (even locally), and then none of the loggers would be able to find the images and thus not upload any. 

This would apply a single fix to all loggers. This should work as the loggers are supposed to check for image existence before logging, i.e. on L115 the check `if f.exists()`:
https://github.com/ultralytics/yolov5/blob/b65ee7fccf901ce636699ec054cb67f0cf2bc6de/utils/loggers/__init__.py#L113-L116

**Comment by ameicler** [comment]
Thanks for your answers!

@AyushExel, indeed, I would need to also eliminate the actual saving of all images to be compliant.

@glenn-jocher, interesting shortcut. So adding a condition L113 if not --nomedia is what you suggest as a fix?

**Reply by glenn-jocher** [reply]
@ameicler no, the idea is the images would never be saved in the first place, the above code I pasted requires no changes.

**Reply by ameicler** [reply]
@glenn-jocher, cool, I see, and how can we prevent to save the image locally in the first place?

**Comment by AyushExel** [comment]
@glenn-jocher @ameicler I've made a PR that enables a `--nomedia` flag which prevents the logging of all media locally and in W&B dash
https://github.com/ultralytics/yolov5/pull/7534

**Reply by ameicler** [reply]
Thank you!

**Comment by glenn-jocher** [comment]
@ameicler good news 😃! Your original issue may now be fixed ✅ in PR #7534 by @AyushExel and myself. This PR adds a `--noplots` flag to train.py that completely suppresses the savings of any media, both locally and in remote YOLOv5 logging integrations like W&B. This means that no sensetive information from your dataset will ever be saved, i.e. no images and no class names are saved or used in metrics or logs. 

Usage example and expected output in runs/train/exp directory are below:

```bash
python train.py --noplots
```

<img width="434" alt="Screen Shot 2022-04-22 at 2 59 52 PM" src="https://user-images.githubusercontent.com/26833433/164801943-03ef6519-e064-4da5-9716-069a9c7811a7.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
